zapcore/json_encoder.go

```diff
@@ -159,20 +144,17 @@ func (enc *jsonEncoder) AppendBool(val bool) {
 }
 
 func (enc *jsonEncoder) Clone() Encoder {
-	clone := jsonPool.Get().(*jsonEncoder)
-	clone.truncate()
-	clone.bytes = append(clone.bytes, enc.bytes...)
-	clone.JSONConfig = enc.JSONConfig
+	clone := &jsonEncoder{JSONConfig: enc.JSONConfig}
+	clone.bytes = append(buffers.Get(), enc.bytes...)
 	return clone
 }
 
-func (enc *jsonEncoder) WriteEntry(sink io.Writer, ent Entry, fields []Field) error {
-	if sink == nil {
-		return errNilSink
+func (enc *jsonEncoder) EncodeEntry(ent Entry, fields []Field) ([]byte, error) {
+	final := &jsonEncoder{
+		JSONConfig: enc.JSONConfig,
+		bytes:      buffers.Get(),
 	}
 
-	final := jsonPool.Get().(*jsonEncoder)
-	final.truncate()
 	final.bytes = append(final.bytes, '{')
 	enc.LevelFormatter(ent.Level).AddTo(final)
 	enc.TimeFormatter(ent.Time).AddTo(final)
```

Comment:
Just for sanity, we should probably use `final.LevelFormatter`. Same for the time and message formatters.